### PR TITLE
Image collage generation and saving

### PR DESCRIPTION
**Pull request by dman926**
Adds the `/create-sponsor-collage` POST endpoint to generate collages with sponsor images.

Tests the implementation in the frontend

### CODE REVIEW

File: api/src/resources/japanese-goblin-emoji-2.png
Comment (dman926) [inline api/src/resources/japanese-goblin-emoji-2.png:1]: Why is this one so good though?

File: api/src/routes/create-sponsor-collage.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+  // Generate array of numbers from 1 to sponsorCount
+  Array.from({ length: sponsorCount }, (_, index) => index + 1).map(
+    (sponsorSuffix) =>
+      pathResolve(
+        __dirname,
+        '..',
+        'resources',
+        `japanese-goblin-emoji-${sponsorSuffix}.png`,
+      ),
+  );
```
Comment (dman926) [inline api/src/routes/create-sponsor-collage.ts:22]: This is temporary for now until there is a more established way of selecting images.

File: api/package.json [hunk 2]
```diff
@@ -4,7 +4,8 @@
   "description": "",
   "main": "dist/index.js",
   "scripts": {
-    "build": "rimraf dist && tsc",
+    "build": "rimraf dist && tsc && npm run copy-resources",
```
Comment (john-rice) [inline api/package.json:7]: rimraf: "The [UNIX command](http://en.wikipedia.org/wiki/Rm_(Unix)) rm -rf for node."

**Review by ZaxAtax — APPROVED** [review]
Japanese Globlin

File: api/src/routes/create-sponsor-collage.ts (hunk 1 above)
Comment (john-rice) [reply api/src/routes/create-sponsor-collage.ts:22]: Sponsored by  japanese-goblin-emoji.

**Review by john-rice — COMMENTED** [review]
Good effort towards adding image processing. While the implementation effectively addresses the unique requirements, should we considering breaking out general use template code from examples and specific use cases?  An approach of keeping general purpose code from specific use cases might help in maintaining the generality of the template while accommodating the specific needs. It is worth exploring the modularization of this feature to enhance the template’s flexibility. What is the generic service that fits the template and what is the implementation? How would this repo be organized in that case? Are the specific use case good to have in this repo or would one or more sample repos that build on this repo be a good organization?

File: api/src/routes/create-sponsor-collage.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+export const createSponsorImageHandler: RouteHandlerMethod = async (
```
Comment (john-rice) [inline api/src/routes/create-sponsor-collage.ts:24]: Is there code that is generic here that could be extracted? Can the sponsor use case be broken out to utilize a refactored abstract / reusable bit of this logic?

Comment (dman926) [reply api/src/routes/create-sponsor-collage.ts:24]: That would be the goal, similarly to what I did to the upload-image endpoint in this PR. This is just to get it in and useable, and avoid having a huge PR.

File: api/src/routes/create-sponsor-collage.ts [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+            return {
+              input: resizedSponsorBuffer,
+              top: Math.round(top),
+              left: Math.round(left),
+            };
+          }),
+        ),
+      )
+      .toBuffer();
```
Comment (dman926) [inline api/src/routes/create-sponsor-collage.ts:131]: @john-rice this is the reusable piece for adding the sponsor images (collaging).

**Pull request merged**